code/modules/reagents/Chemistry-Grinder.dm

```diff
@@ -267,16 +248,15 @@
 	addtimer(CALLBACK(src, .proc/shake, user, 40), 0)
 
 /obj/machinery/reagentgrinder/proc/shake(mob/user, duration)
-	for(var/i = 1, i<=duration, i++)
-		sleep(1)
-		if(!user || !Adjacent(user))
-			break
-		if(user.is_jittery)
-			continue
+	if(!user)
+		return FALSE
+	if((!Adjacent(user) || duration <= 0) && !user.is_jittery)
+		user.do_jitter(0) //resets the icon.
+		return FALSE
+	if(!user.is_jittery)
 		user.do_jitter(4)
 
-	if(user && !user.is_jittery)
-		user.do_jitter(0) //resets the icon.
+	addtimer(CALLBACK(src, .proc/shake, user, --duration), 1, TIMER_UNIQUE)
```

Comment:
Is this really worth it?

Reply:
Probably not. Changed back to a version that uses `sleep()`